### PR TITLE
CHORE: update error handling to not throw lock error

### PR DESCRIPTION
**Pull request by brendanbond**
## Link to Jira Ticket

n/a

## Description

Upon a db related error, the server will attempt to unlock. This will fail in the case of e.g. database connection errors, because there is no database to unlock. This PR updates error handling to throw the correct error rather than always throwing the unlock error. 

## Breaking Changes / Backwards Compatibility

n/a

## Dependencies

n/a

## How has this PR been tested?

n/a

## Checklist:

- [x] I have completed the above PR template
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] My changes generate no new warnings
- [ ] My changes include tests that prove my fix is effective (or that my feature works as intended)
- [ ] New and existing unit/integration tests pass locally with my changes
- [ ] Any dependent changes have corresponding PRs that are listed above

### CODE REVIEW

File: src/db/index.js [hunk 1]
```diff
@@ -744,7 +749,7 @@ module.exports = function(formio) {
         await unlock();
         return db;
       }
-      catch (err) {
+      catch (ignoreErr) {
```
Comment (mikekotikov) [inline src/db/index.js:752]: You probably didn't want to rename that one

**Review by mikekotikov — APPROVED** [review]
LGTM, just please make sure to rename the second `ignoreErr` back to `err`

File: src/db/index.js (hunk 1 above)
Comment (brendanbond) [reply src/db/index.js:752]: actually that was intentional - if an error happens with the database, there will be no lock to unlock, and `await unlock()` will throw an error - we want to surface the _original_ error, not the unlock error (so we can safely ignore the unlock error in other words)

Comment (brendanbond) [reply src/db/index.js:752]: in other words, we want to throw the original `err` and not shadow that variable with the unlock error

**Pull request merged**